### PR TITLE
build API docs for 0.17

### PR DESCRIPTION
**Pull request by RichieEscarez**
Update API ref docs using new 0.17 knative components

https://github.com/knative/docs/tree/master/docs/reference#updating-api-reference-docs-for-knative-maintainers

```
cd hack

KNATIVE_SERVING_COMMIT=v0.17.0 KNATIVE_EVENTING_COMMIT=v0.17.0 KNATIVE_EVENTING_CONTRIB_COMMIT=v0.17.0 ./gen-api-reference-docs.sh
```

### CODE REVIEW

File: docs/reference/serving.md [hunk 1]
```diff
@@ -1165,7 +920,7 @@ ServiceStatus
 <p>
 (<em>Appears on:</em>
 <a href="#serving.knative.dev/v1beta1.Configuration">Configuration</a>, 
-<a href="#serving.knative.dev/v1.Configuration">Configuration</a>,
+<a href="#serving.knative.dev/v1.Configuration">Configuration</a>,
```
Comment (mattmoor) [inline docs/reference/serving.md:923]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1.Configuration">Configuration</a>,
```

File: docs/reference/serving.md [hunk 2]
```diff
@@ -1199,8 +954,8 @@ RevisionTemplateSpec
 </h3>
 <p>
 (<em>Appears on:</em>
-<a href="#serving.knative.dev/v1.Configuration">Configuration</a>, 
-<a href="#serving.knative.dev/v1beta1.Configuration">Configuration</a>)
+<a href="#serving.knative.dev/v1beta1.Configuration">Configuration</a>,
```
Comment (mattmoor) [inline docs/reference/serving.md:957]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1beta1.Configuration">Configuration</a>,
```

File: docs/reference/serving.md [hunk 3]
```diff
@@ -1334,8 +1089,8 @@ string
 </h3>
 <p>
 (<em>Appears on:</em>
-<a href="#serving.knative.dev/v1.Revision">Revision</a>, 
 <a href="#serving.knative.dev/v1beta1.Revision">Revision</a>, 
+<a href="#serving.knative.dev/v1.Revision">Revision</a>,
```
Comment (mattmoor) [inline docs/reference/serving.md:1093]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1.Revision">Revision</a>,
```

File: docs/reference/serving.md [hunk 4]
```diff
@@ -1400,8 +1155,8 @@ be provided.</p>
 </h3>
 <p>
 (<em>Appears on:</em>
-<a href="#serving.knative.dev/v1.Revision">Revision</a>, 
-<a href="#serving.knative.dev/v1beta1.Revision">Revision</a>)
+<a href="#serving.knative.dev/v1beta1.Revision">Revision</a>,
```
Comment (mattmoor) [inline docs/reference/serving.md:1158]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1beta1.Revision">Revision</a>,
```

File: docs/reference/serving.md [hunk 5]
```diff
@@ -1595,8 +1350,8 @@ be provided.</p>
 </h3>
 <p>
 (<em>Appears on:</em>
-<a href="#serving.knative.dev/v1.Route">Route</a>, 
 <a href="#serving.knative.dev/v1beta1.Route">Route</a>, 
+<a href="#serving.knative.dev/v1.Route">Route</a>,
```
Comment (mattmoor) [inline docs/reference/serving.md:1354]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1.Route">Route</a>,
```

File: docs/reference/serving.md [hunk 6]
```diff
@@ -1631,8 +1386,8 @@ revisions and configurations.</p>
 </h3>
 <p>
 (<em>Appears on:</em>
-<a href="#serving.knative.dev/v1.Route">Route</a>, 
-<a href="#serving.knative.dev/v1beta1.Route">Route</a>)
+<a href="#serving.knative.dev/v1beta1.Route">Route</a>,
```
Comment (mattmoor) [inline docs/reference/serving.md:1389]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1beta1.Route">Route</a>,
```

File: docs/reference/serving.md [hunk 7]
```diff
@@ -1739,12 +1494,17 @@ LatestReadyRevisionName that we last observed.</p>
...
+<p>
+<p>RoutingState represents states of a revision with regards to serving a route.</p>
+</p>
 <h3 id="serving.knative.dev/v1.ServiceSpec">ServiceSpec
 </h3>
 <p>
 (<em>Appears on:</em>
-<a href="#serving.knative.dev/v1.Service">Service</a>, 
-<a href="#serving.knative.dev/v1beta1.Service">Service</a>)
+<a href="#serving.knative.dev/v1beta1.Service">Service</a>,
```
Comment (mattmoor) [inline docs/reference/serving.md:1506]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1beta1.Service">Service</a>,
```

File: docs/reference/serving.md [hunk 8]
```diff
@@ -1804,8 +1564,8 @@ defaults).</p>
...
```
Comment (mattmoor) [inline docs/reference/serving.md:1567]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1beta1.Service">Service</a>,
```

File: docs/reference/serving.md (hunk 1 above)
Comment (mattmoor) [inline docs/reference/serving.md:923]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1.Configuration">Configuration</a>,
```

File: docs/reference/serving.md (hunk 3 above)
Comment (mattmoor) [inline docs/reference/serving.md:1093]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1.Revision">Revision</a>,
```

File: docs/reference/serving.md (hunk 4 above)
Comment (mattmoor) [inline docs/reference/serving.md:1158]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1beta1.Revision">Revision</a>,
```

File: docs/reference/serving.md (hunk 5 above)
Comment (mattmoor) [inline docs/reference/serving.md:1354]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1.Route">Route</a>,
```

File: docs/reference/serving.md (hunk 6 above)
Comment (mattmoor) [inline docs/reference/serving.md:1389]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1beta1.Route">Route</a>,
```

File: docs/reference/serving.md (hunk 7 above)
Comment (mattmoor) [inline docs/reference/serving.md:1506]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1beta1.Service">Service</a>,
```

File: docs/reference/serving.md (hunk 8 above)
Comment (mattmoor) [inline docs/reference/serving.md:1567]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1beta1.Service">Service</a>,
```

File: docs/reference/serving.md (hunk 3 above)
Comment (mattmoor) [inline docs/reference/serving.md:1093]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1.Revision">Revision</a>,
```

File: docs/reference/serving.md (hunk 4 above)
Comment (mattmoor) [inline docs/reference/serving.md:1158]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1beta1.Revision">Revision</a>,
```

File: docs/reference/serving.md (hunk 5 above)
Comment (mattmoor) [inline docs/reference/serving.md:1354]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1.Route">Route</a>,
```

File: docs/reference/serving.md (hunk 6 above)
Comment (mattmoor) [inline docs/reference/serving.md:1389]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1beta1.Route">Route</a>,
```

File: docs/reference/serving.md (hunk 7 above)
Comment (mattmoor) [inline docs/reference/serving.md:1506]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1beta1.Service">Service</a>,
```

File: docs/reference/serving.md (hunk 8 above)
Comment (mattmoor) [inline docs/reference/serving.md:1567]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1beta1.Service">Service</a>,
```

File: docs/reference/serving.md (hunk 4 above)
Comment (mattmoor) [inline docs/reference/serving.md:1158]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1beta1.Revision">Revision</a>,
```

File: docs/reference/serving.md (hunk 5 above)
Comment (mattmoor) [inline docs/reference/serving.md:1354]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1.Route">Route</a>,
```

File: docs/reference/serving.md (hunk 6 above)
Comment (mattmoor) [inline docs/reference/serving.md:1389]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1beta1.Route">Route</a>,
```

File: docs/reference/serving.md (hunk 7 above)
Comment (mattmoor) [inline docs/reference/serving.md:1506]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1beta1.Service">Service</a>,
```

File: docs/reference/serving.md (hunk 8 above)
Comment (mattmoor) [inline docs/reference/serving.md:1567]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1beta1.Service">Service</a>,
```

File: docs/reference/serving.md (hunk 5 above)
Comment (mattmoor) [inline docs/reference/serving.md:1354]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1.Route">Route</a>,
```

File: docs/reference/serving.md (hunk 6 above)
Comment (mattmoor) [inline docs/reference/serving.md:1389]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1beta1.Route">Route</a>,
```

File: docs/reference/serving.md (hunk 7 above)
Comment (mattmoor) [inline docs/reference/serving.md:1506]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1beta1.Service">Service</a>,
```

File: docs/reference/serving.md (hunk 8 above)
Comment (mattmoor) [inline docs/reference/serving.md:1567]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1beta1.Service">Service</a>,
```

File: docs/reference/serving.md (hunk 6 above)
Comment (mattmoor) [inline docs/reference/serving.md:1389]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1beta1.Route">Route</a>,
```

File: docs/reference/serving.md (hunk 7 above)
Comment (mattmoor) [inline docs/reference/serving.md:1506]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1beta1.Service">Service</a>,
```

File: docs/reference/serving.md (hunk 8 above)
Comment (mattmoor) [inline docs/reference/serving.md:1567]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1beta1.Service">Service</a>,
```

File: docs/reference/serving.md (hunk 7 above)
Comment (mattmoor) [inline docs/reference/serving.md:1506]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1beta1.Service">Service</a>,
```

File: docs/reference/serving.md (hunk 8 above)
Comment (mattmoor) [inline docs/reference/serving.md:1567]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1beta1.Service">Service</a>,
```

Comment (mattmoor) [inline docs/reference/serving.md:1567]: Remove trailing whitespace:
```suggestion
<a href="#serving.knative.dev/v1beta1.Service">Service</a>,
```

**Comment by RichieEscarez** [comment]
all suggestions applied (i wonder if we can get the script to remove all the whitespace automatically?)

just need an /lgtm to merge

**Pull request merged**